### PR TITLE
fix(content-group-horizontal): multiple style bugs

### PR DESCRIPTION
**Pull request by NSDrowned**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5870

### Description

- [x] Make sure the spacing after the section heading is 32px as according to AEM recommendations
- [x] There is an extra bottom padding for the links that makes the spacing between them and the horizontal rule 64px + an extra 16px
- [x] The grid alignment is off. Content starts at col 4
- [x] Horizontal rule should span 12 col and overhang
- [ ] Adjust the content item width on md

### CODE REVIEW

**Comment by kennylam** [comment]
@NSDrowned Since this component is being deprecated in favor of the new `content-block-horizontal`, you should wait until #6489 is merged so you can apply these changes to the new files.

**Comment by NSDrowned** [comment]
@kennylam got it! 👍

**Comment by jeffchew** [comment]
@NSDrowned the deprecation PR is merged now so you can update your changes here.

File: packages/styles/scss/components/content-group-horizontal/_content-group-horizontal.scss
Comment (kennylam) [inline packages/styles/scss/components/content-group-horizontal/_content-group-horizontal.scss:19]: Where is this being applied? And why would such a large value be needed on a margin?

File: packages/styles/scss/components/content-block-horizontal/_content-block-horizontal.scss
Comment (kennylam) [inline packages/styles/scss/components/content-block-horizontal/_content-block-horizontal.scss:18]: Where is this being applied? And why would such a large value be needed on a margin?

**Review by kennylam — COMMENTED** [review]
Thanks for this @NSDrowned. I can't see where the negative margins are being applied on the horizontal rules.

**Comment by shixiedesign** [comment]
A word on the horizontal rule / border below the component - from the visual side, the rule should be 12 col, matching other block level components like [Content block simple](https://ibmdotcom-web-components-canary.mybluemix.net/?path=/story/components-content-block-simple--default), where the rule starts in gutter of col 5 and goes all the way to col 16. I'm not sure what's the correct way to achieve this though. 

Let me know when it's ready to be reviewed again! Thanks!

File: packages/styles/scss/components/content-group-horizontal/_content-group-horizontal.scss
Comment (NSDrowned) [reply packages/styles/scss/components/content-group-horizontal/_content-group-horizontal.scss]: Talked with @shixiedesign and component is now 12 columns wide. Content (except for the HR) will take up 10 columns.

File: packages/styles/scss/components/content-block-horizontal/_content-block-horizontal.scss
Comment (NSDrowned) [reply packages/styles/scss/components/content-block-horizontal/_content-block-horizontal.scss]: This has been removed 👍

**Review by kennylam — APPROVED** [review]
LGTM thanks @NSDrowned!

**Comment by shixiedesign** [comment]
Signing off for a few days and just leaving the last comment here: 

@NSDrowned as we discussed on Slack, 

1. `Content group horizontal` and `Content block horizontal` should match exactly. Sorry about the duplication. In a few more releases, we would remove Content group horizontal completely and there won't be duplicated code then.

2. `Content BLOCK horizontal` has the correct bottom rule. Need to change `content GROUP horizontal`

3. `Content GROUP horizontal` has the correct width for content. Need to change `content BLOCK horizontal`. 

Final outcome should look like 
image

**Comment by DragosRistici** [comment]
Hey, @NSDrowned, a few little things:

1. Spacing above and below heading on md should also be 32px:

Screenshot 2021-07-21 at 16 40 32
Screenshot 2021-07-21 at 16 40 43

2. Spacing between the CTAs and horiz rule should be 64px. And the spacing between the short copy and the CTAs is MINIMUM 64px. This means that its variable so the min-height of 296px of the content item applies.
Screenshot 2021-07-21 at 16 54 32
Screenshot 2021-07-21 at 16 41 42

3.  Last element's spacing before the horizontal rule should be 160. Now the 160 is AFTER the rule. 
Screenshot 2021-07-21 at 16 45 23
Screenshot 2021-07-21 at 17 32 10
4. Here before the last horizontal rule the spacing should be 96.
Screenshot 2021-07-21 at 18 10 24

5. for mobile, adjust spacing also
Screenshot 2021-07-21 at 18 12 12
Screenshot 2021-07-21 at 18 11 37

6. Also last horiz rule spacing should be 64.
Screenshot 2021-07-21 at 18 13 07
Screenshot 2021-07-21 at 18 13 50

**Comment by jeffchew** [comment]
Just checking @DragosRistici @NSDrowned , in the NextJS test app the content group horizontal shifted a bit (which I think might be part of the fixes here?), but also noticing some alignment issues.

image

**Comment by jeffchew** [comment]
@NSDrowned this is quite stale, anything needed to move this along?

**Comment by NSDrowned** [comment]
> @NSDrowned this is quite stale, anything needed to move this along?

Just waiting for review. Content shift was part of the changes. Lots of movement.

**Comment by DragosRistici** [comment]
Looking almost ready. 

Little things:

1. If a content item horizontal has 1 row of copy on the right side, it loses grid. 
Screenshot 2021-07-26 at 17 32 17

2. This unused h4 creates some problems: it adds extra space when items are stacked, and also it looks like it creates one accessibility issue:
 Screenshot 2021-07-26 at 18 19 26
Screenshot 2021-07-26 at 18 21 29
Screenshot 2021-07-26 at 18 22 25

**Comment by NSDrowned** [comment]
@DragosRistici nice catch 👍  although these bugs are not related to this component they do affect it. I made the heading in `link-list` only render if it actually has a value, which fixes the empty H4 violation. The other bug was actually happening in `content-item-horizontal`. It was missing a 100% width for the parent which would cause the first column to shrink. Both fixed now 👍

**Comment by DragosRistici** [comment]
Hey, @NSDrowned you are almost there. Please, take care of these little style details that are not yet resolved for the smaller breakpoints. If there's one of them that needs a new issue, let me know. Let's close this today. 

### 1. Medium breakpoint

#### This is the spec:
Screenshot 2021-07-28 at 11 11 29

#### This is what I still see:
a. Top spacing should be 32px
Screenshot 2021-07-28 at 11 13 56
b. Spacing under heading should be 32px
Screenshot 2021-07-28 at 11 14 01
c. Spacing at the end of the element should be 64px (for `Web Components`)
Screenshot 2021-07-28 at 11 14 15

d. the CTA container spacings on `React` are a bit off. I see a 27px now, a 49px, and the 1 link adds a lot of space due to this `margin-top:auto` added to the CTA container. 
Screenshot 2021-07-28 at 11 35 37
Screenshot 2021-07-28 at 11 34 45
Screenshot 2021-07-28 at 11 33 59
Screenshot 2021-07-28 at 11 33 39
Screenshot 2021-07-28 at 11 33 31




### 2. Small breakpoint

This is the spec: 
Screenshot 2021-07-28 at 11 21 01


#### This is what I see:
a.  Top spacing should be 32px
Screenshot 2021-07-28 at 11 19 05

b. Bottom spacing of an item should be 32px. Now, for `React`, you got 16 + 32
Screenshot 2021-07-28 at 11 20 07Screenshot 2021-07-28 at 11 20 01


## *Important: Please, let me know if any of this should be in a new issue and I will create a new one.

**Comment by shixiedesign** [comment]
Hey @NSDrowned is this ready for review again?

**Comment by NSDrowned** [comment]
@shixiedesign, yes it is! 👍

**Comment by DragosRistici** [comment]
I've reviewed it again with @NSDrowned on Slack. Minor spacing things and it should be ready to go.

**Comment by jeffchew** [comment]
@shixiedesign @DragosRistici the alignment on NextJS test app appears to be fixed now:

image

**Review by DragosRistici — APPROVED** [review]
LGTM, thank you @NSDrowned!

**Pull request merged**
